### DISCUSSION TITLE
Project Roadmap

### DISCUSSION THREAD

**Discussion by webfansplz** [vuejs/devtools#112] [General]
- [x] Fully compatible with V6 plugin API
- [x] Refactor Messaging Channel
- [x] Ship Browser Extension
- [x] Timeline Panel
- [ ] Documentation
   - [ ]  DevTools packages diagrams
   - [ ]  Contribution guide
   - [x]  Migration Guide
- [x] Integration features into Nuxt DevTools
   - [x] Pinia panel
   - [x] Components panel
- [x]  Applet package
   - [x] Reuse core modules/utils with Nuxt DevTools  
- [ ] Reactivity Tracing feature

**Comment by alexzhang1030** [comment]
Please be aware that the new `vite-plugin-vue-devtools` have rewritten all the code, and the functionality will differ compared to the old `vite-plugin-vue-devtools`. We will migrate gradually.

**Comment by Mister-Hope** [comment]
I am a bit confused about the project roadmap now, The project is not actively maintained in the past 45 days. But I see there are still plenty of TODOs at this page.

No browser extension releases, but https://www.npmjs.com/package/@vue/devtools-api is using v7 as latest (which I offten upgrades to by mistake.)


So my biggest concern is about the conflict between:

- latest tag with v7, not actively maintained
- Missing roadmaps and no devtool extension (not pretty available for users)

A explanation is appreciated, or at least we may set latest tag back to v6?

Under my understanding, if authors like me want to develop a plugin for the devtool, I shall still publish it with v6. Users are definitely not expected to see a guideline telling them to install the new v7 globally and launch it every time manualy for debugging.

**Comment by webfansplz** [comment]
> I am a bit confused about the project roadmap now, The project is not actively maintained in the past 45 days. But I see there are still plenty of TODOs at this page.
> 
> No browser extension releases, but https://www.npmjs.com/package/@vue/devtools-api is using v7 as latest (which I offten upgrades to by mistake.)
> 
> So my biggest concern is about the conflict between:
> 
> * latest tag with v7, not actively maintained
> * Missing roadmaps and no devtool extension (not pretty available for users)
> 
> A explanation is appreciated, or at least we may set latest tag back to v6?
> 
> Under my understanding, if authors like me want to develop a plugin for the devtool, I shall still publish it with v6. Users are definitely not expected to see a guideline telling them to install the new v7 globally and launch it every time manualy for debugging.

First, I'd like to clarify the maintenance status of the project. We are still actively maintaining it, but we don't have full-time maintainers, we have other work to do, and I appreciate the time and effort that contributors put into it.

Regarding browser extensions, we need to deal with compatibility issues with older versions of the original. Because older versions support Vue2, and there are a lot of users out there. We need to have a good plan to deal with them, and that will take some time.

Regarding the Vue DevTools Plugin API, we are trying to be fully backward compatible with the V6 API to facilitate compatibility with V6 plugins in the ecosystem.

Finally, thanks for your feedback, I have rolled back the latest tag of `@vue/devtools-api` to v6, hope it will help you.

**Comment by paulmelero** [comment]
Hi, @webfansplz. First of all, thanks a lot for the effort you and other contributors put into the new devtools! I just tried them inside a Storybook project and they work magnificently. I wanted to ask if there are plans to add Vue component events in the UI too. I'd love to help as well!

**Reply by alexzhang1030** [reply]
Absolutely! PRs are very welcome!

**Comment by paulmelero** [comment]
Another question, do you plan to add something like `@nuxt/devtools-kit` so users can build and register their own tabs?

As seen in: https://devtools.nuxt.com/module/guide

**Reply by alexzhang1030** [reply]
Yeah, we have a simple `plugin-api` now, we will consider making this API more powerful.

**Reply by webfansplz** [reply]
> Another question, do you plan to add something like `@nuxt/devtools-kit` so users can build and register their own tabs?
> 
> As seen in: https://devtools.nuxt.com/module/guide

Hi, we made the v6 plugin API backward compatible in v7.3.0, and now users can customize tabs through it. (see: https://devtools-next.vuejs.org/plugins/api)

**Comment by kazupon** [comment]
Hello!
Thank you for your amazing project!
I have a question.

Recently `addCustomTab` has been supported. Its API supports iframe, so 3rd tools gives us more freedom for debug tools.
https://devtools-next.vuejs.org/plugins/api#addcustomtab

Currently, I seems that it does not support RPC, are there any plans to do so in the future?

I know from reading the source code that `vite-plugin-insect` is integrated using `addCustomTab`.
`vite-plugin-insect` uses vite's RPC and negociates with its vite dev server and browser client.

I hope that support the RPC between the iframe side and on the vue devtools side.

**Reply by webfansplz** [reply]
I think supporting RPC messaging for `addCustomTab` API makes sense. Let's track it in #537

**Comment by zhangenming** [comment]
what is `Reactivity Tracing feature`
Visual the vue Reactivity dependency(ref/computed/watch...) graph ?

![1724839536256](https://github.com/user-attachments/assets/b2ac0d6d-2a0c-4ad0-8d15-460b35c2d6f9)
![1724839543888](https://github.com/user-attachments/assets/f6327358-5d8c-48c5-bccd-4ca00e57e412)

**Reply by alexzhang1030** [reply]
We have been thinking about this feature for a long time. The hard point is correctly analyzing the reactivity dependencies (especially for cross files/modules).  The runtime analyzer will be great, but it cannot analyze the location in the file of the variable.

We may implement this in the future by combining runtime and compile-time. If you are interested in this, we can have a more in-depth discussion :)

**Reply by matthiasPOE** [reply]
@alexzhang1030 Im looking for this: https://github.com/vuejs/devtools-v6/issues/862 is this already in here or does it relate to the tracing?

**Comment by s3xybun** [comment]
Reactivity Tracing is my favorite feature `(in vue2? I have memories of using it)`, which is really useful. Hope it could come back soon!

**Comment by YY-Nexus** [comment]
> * [x] 完全兼容V6插件API
> * [x] 重构消息通道
> * [x] 船舶浏览器扩展
> * [x] 时间轴面板
> * [ ] 文档
>   
>   * [ ]   DevTools 软件包图表
>   * [ ]   贡献指南
>   * [x]   迁移指南
> * [x] 将功能集成到 Nuxt DevTools
>   
>   * [x] 松属植物面板
>   * [x] 元件面板
> * [x]   小程序包
>   
>   * [x] 使用 Nuxt DevTools 重用核心模块/实用程序
> * [ ] 反应性追踪功能
